mypy_django_plugin/main.py

```diff
@@ -111,15 +111,18 @@ def get_additional_deps(self, file: MypyFile) -> List[Tuple[int, str, int]]:
             return [self._new_dependency("typing"), self._new_dependency("django_stubs_ext")]
 
         # for `get_user_model()`
-        if self.django_context.settings:
-            if file.fullname == "django.contrib.auth" or file.fullname in {"django.http", "django.http.request"}:
-                auth_user_model_name = self.django_context.settings.AUTH_USER_MODEL
-                try:
-                    auth_user_module = self.django_context.apps_registry.get_model(auth_user_model_name).__module__
-                except LookupError:
-                    # get_user_model() model app is not installed
-                    return []
-                return [self._new_dependency(auth_user_module), self._new_dependency("django_stubs_ext")]
+        if self.django_context.settings and any(
+            i.id == "django.contrib.auth" and any(name == "get_user_model" for name, _ in i.names)
+            for i in file.imports
+            if isinstance(i, ImportFrom)
+        ):
```

Comment:
Isn't this a regular import dependency, on `django.contrib.auth`? Which mypy should establish by itself

At least I was thinking this hook is for "invisible" dependencies

Reply:
This adds a dependency on the configured user model, which might not be in `django.contrib.auth`. I guess one option we could do is to add the user model as a dependency of `django.contrib.auth`, that might be cleaner?

Reply:
For some reason, doing it like that causes the same issue as above where mypy gets stuck in an infinite loop